src/Internal/Payment/State/InitialState.php

```diff
@@ -120,18 +122,25 @@ public function start_processing( PaymentRequest $request ) {
 		}
 		// End multiple verification checks.
 
-		// Payments are currently based on intents, request one from the API.
+		// Payments are currently based on intents, request one from the API, and before that make sure that customer is created or updated.
 		try {
-			$context = $this->get_context();
-			$intent  = $this->payment_request_service->create_intent( $context );
+			$context  = $this->get_context();
+			$order_id = $context->get_order_id();
+
+			// Create or update customer and customer details.
+			$customer_id = $this->customer_service->get_or_create_customer_id_from_order( $context->get_user_id(), $this->order_service->_deprecated_get_order( $order_id ) );
```

Comment:
Let's make this one multi-line:


```suggestion
			$customer_id = $this->customer_service->get_or_create_customer_id_from_order(
				$context->get_user_id(),
				$this->order_service->_deprecated_get_order( $order_id )
			);
```

It's by far the longest line in this file.